include/desktop/toplevel.h

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c), Niclas Meyer <niclas@countingsort.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef KIWMI_DESKTOP_TOPLEVEL_H
+#define KIWMI_DESKTOP_TOPLEVEL_H
+
+struct wlr_surface;
+struct kiwmi_desktop;
+
+enum kiwmi_toplevel_type {
+    KIWMI_TOPLEVEL_VIEW,
+    KIWMI_TOPLEVEL_LAYER,
+};
+
+// prolly has to be initialized by the specific type of toplevel?
+struct kiwmi_toplevel {
+    struct wlr_scene_tree *scene_subtree;
+    struct wlr_scene_tree *popups_scene_subtree;
+    // in the future, scene_subtree will contain more than just this
+    struct wlr_scene_node *surface_scene_node;
```

Comment:
I seem to have issues finding the sweet point between clarity & brevity. Better names?